src/model/table.py

```diff
@@ -55,18 +81,19 @@ def __get_all_primary_keys__(self):
 
     def get_by_primary_key(self, primary_key):
         path = self.__get_primary_key_path__(primary_key)
-        if not os.path.exists(path):
+        if path is None or not os.path.isfile(path):
```

Comment:
You shouldn’t read the row file here. You should read it from Row class. Because the Row class is the place that write the file.

You should have Row.load_by_primary_key(primary_key)

Reply:
Yo can remove this method. load by primary key return null in case file doesn’t exists